src/HivePlot/HivePlot.tsx

```diff
@@ -378,47 +809,61 @@ export class HivePlot extends Component<HivePlotProps, HivePlotState> {
             reference={tooltipReference}
             placement={placement}
             content={() =>
-              formatter(axis, nodes, link, node) ||
-              (node ? (
-                <HiveTooltip node={node} nodes={nodes} axis={axis} />
+              formatter(axis, nodes, linkTooltipData, nodeTooltipData) ||
+              (nodeTooltipData ? (
+                <HiveTooltip node={nodeTooltipData} nodes={nodes} axis={axis} />
               ) : null)
             }
           />
         )}
       </Fragment>
     );
-  }
-
-  renderChart({ height, width }) {
-    const { innerRadius, axis, colorScheme, label, className } = this.props;
-
-    const data = this.prepareData({
-      dimension: Math.min(height, width),
-      innerRadius,
-      colorScheme,
-      axis,
-      label
-    });
-
-    return (
-      <Fragment>
-        <svg width={width} height={height} className={classNames(className)}>
-          <g transform={`translate(${width / 2}, ${height / 2 + innerRadius})`}>
-            {this.renderAxis(data)}
-            {this.renderLinks(data)}
-            {this.renderNodes(data)}
-          </g>
-        </svg>
-        {this.renderTooltip()}
-      </Fragment>
-    );
-  }
-
-  render() {
-    return (
-      <ChartContainer height={this.props.height} width={this.props.width}>
-        {this.renderChart.bind(this)}
-      </ChartContainer>
-    );
-  }
-}
+  }, [
+    tooltip,
+    disabled,
+    axis,
+    nodes,
+    active,
+    tooltipReference,
+    linkTooltipData,
+    nodeTooltipData
+  ]);
+
+  const renderChart = useCallback(
+    ({ height: containerHeight, width: containerWidth }) => {
+      const data = prepareData({
+        dimension: Math.min(containerHeight, containerWidth),
+        innerRadius,
+        colorScheme,
+        axis,
+        label
+      });
+      return (
+        <Fragment>
+          <svg
```

Comment:
I could be wrong but doesn't the chart container do this for us? This is really old code and prob needs a few updates.